.prettierignore

```diff
@@ -1,14 +1,33 @@
-/packages/chevrotain/lib
-/packages/chevrotain/dev
+### From .gitignore
+/.idea
+node_modules
+/.idea/dictionaries/
+/.idea/
+/.tscache
+.baseDir.ts
+/bower_components
+/npm-debug.log
+yarn-error.log
+.DS_Store
+
+# Chevrotain packages ignores
+# as prettier --ignore-path does not support nested folders ignores
+/packages/website/docs/.vuepress/dist/
+/packages/types/dev
+/packages/**/lib
+/packages/**/coverage
+/packages/**/.nyc_output
+/packages/chevrotain/lib_esm
+/packages/chevrotain/.nyc_output
 /packages/chevrotain/coverage
 /packages/chevrotain/scripts/gh-pages
-/examples/parser/minification/gen/
-/examples/webpack/gen/
-/examples/implementation_languages/typescript/typescript_json.js
-/examples/implementation_languages/coffeescript/coffeescript_json.js
-/packages/chevrotain/docs/.vuepress/dist
-**/node_modules/
+/packages/chevrotain/gh-pages
+
+# Examples ignore
 /examples/webpack/lib
-/greenkeeper.json
-.nyc_output
+/examples/implementation_languages/typescript/*.js
+/examples/implementation_languages/coffeescript/*.js
+/examples/parser/minification/gen/
 
+### Prettier specific
+/packages/dts-gen/test/snapshots/**/output.d.ts
```

Comment:
perhaps instead of duplicating ignore configs in the project
we could format the snapshots output with prettier's **programmatic api** ?

Reply:
so allow the repository prettier to format the output and have a step in the snapshot tests to apply a prettier config to format the expected output before comparing to the actual output?

bit leery of transforming the output before performing the assertions.  since the formatter could be cleaning up some silly formatting errors.  though that could be caught by a test unrelated to the snapshots.  i.e. `options_spec.ts`

Alternatively we could generate the DTS in the same format as the repository instead.  users can then run their own formatters over the output.  did you have a preference on the format we generate?

Reply:
> not sure I follow. do you mean add a guide to generate the DTS? or mention this package in the syntax diagrams docs?

I am confused I meant the CST docs:
- https://github.com/Chevrotain/chevrotain/blob/master/packages/website/docs/guide/concrete_syntax_tree.md

Reply:
is that a response?  or a duplicate post?

Reply:
I replied in the wrong input box :)

Reply:
The goal is to have a simple solution that will enable fast development flows.
I don't have a perfect solution here, just random thoughts.

## Generating output in the same format as the repo.
This is generally good, but may need to be adjusted when `prettier` is updated.

## Perhaps we should avoid comparing the text?

For example if our snapshots would be the serialized typescript AST (e.g from Babel typescript parser)
than we won't care about slight formatting differences

WDYT?

Reply:
I think we should compare the text.  An AST would be harder to read/understand.

I'm leaning towards duplicating the ignore configs at the moment.  We'd avoid having to adjust either the source or the snapshots when prettier is updated.

As an aside, we should probably delete .prettierignore if we're not going to duplicate the ignore configs (since it's not used atm)